iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -28,166 +28,649 @@ using namespace iox::capro;
-    ServiceDescription csd3Eq = ServiceDescription(1, 2, testAnyInstanceID);
-    ServiceDescription csd3Ne = ServiceDescription(9, 2, testAnyInstanceID);
-    ServiceDescription csd4 = ServiceDescription(1, 2, 3);
-    ServiceDescription csd4Eq = ServiceDescription(1, 2, 3);
-    ServiceDescription csd4Ne = ServiceDescription(9, 9, 9);
-    ServiceDescription csdIdStr = ServiceDescription(service1, instance1, event1);
     void SetUp(){};
     void TearDown(){};
 };
 
-TEST_F(ServiceDescription_test, CtorSerial)
+TEST_F(ServiceDescription_test, ServiceDescriptionSerializationCtorCreatesServiceDescriptionWithValuesPassedToTheCtor)
 {
     ServiceDescription::ClassHash testHash = {1, 2, 3, 4};
-    auto serialObj = iox::cxx::Serialization::create(service1.c_str(),
-                                                     instance1.c_str(),
-                                                     event1.c_str(),
-                                                     1,
-                                                     2,
-                                                     3,
+    IdString_t testService{"Service"};
+    IdString_t testInstance{"Instance"};
+    IdString_t testEvent{"Event"};
+    uint16_t testServiceID = 1U;
+    uint16_t testInstanceID = 2U;
+    uint16_t testEventID = 3U;
+    Scope testScope = Scope::INTERNAL;
+    Interfaces testInterfaceSource = Interfaces::INTERNAL;
+    auto serialObj = iox::cxx::Serialization::create(testService.c_str(),
+                                                     testInstance.c_str(),
+                                                     testEvent.c_str(),
+                                                     testServiceID,
+                                                     testInstanceID,
+                                                     testEventID,
                                                      testHash[0],
                                                      testHash[1],
                                                      testHash[2],
                                                      testHash[3],
-                                                     true);
-    ServiceDescription csd = ServiceDescription(serialObj);
-    csd.setInternal();
-    EXPECT_THAT(csd.getServiceIDString(), Eq(service1));
-    EXPECT_THAT(csd.getInstanceIDString(), Eq(instance1));
-    EXPECT_THAT(csd.getEventIDString(), Eq(event1));
-    EXPECT_THAT(csd.getServiceID(), Eq(1));
-    EXPECT_THAT(csd.getInstanceID(), Eq(2));
-    EXPECT_THAT(csd.getEventID(), Eq(3));
-    EXPECT_THAT((csd.getClassHash())[0], Eq(testHash[0]));
-    EXPECT_THAT((csd.getClassHash())[1], Eq(testHash[1]));
-    EXPECT_THAT((csd.getClassHash())[2], Eq(testHash[2]));
-    EXPECT_THAT((csd.getClassHash())[3], Eq(testHash[3]));
-    EXPECT_THAT(csd.hasServiceOnlyDescription(), Eq(true));
-    EXPECT_THAT(csd.isInternal(), Eq(true));
+                                                     true,
```

Comment:
I see this as problem here because we can create a servicedescription through the serialization with the three friends and `hasServiceOnlyDescription()` returns true, which should not be the case.

Reply:
Sorry, I didn't get you. Should there be a refactoring of the existing code or the test ?

Reply:
The problem is that it is possible to set `m_hasServiceOnlyDescription` wrongly.
What if you pass three valid IDs and set `m_hasServiceOnlyDescription` manually to true? Then we have a Servicedescription with a valid event ID but it is treated as reduced ServiceDescription with instance and service id only.